frontend/src/modules/labwork/services.ts

```diff
@@ -0,0 +1,124 @@
+import { FMSId, FMSLabworkSummary } from "../../models/fms_api_models"
+import { LabworkStepGroup, LabworkSummary, LabworkSummaryProtocol, LabworkSummaryStep } from "../../models/labwork_summary"
+
+/**
+ * Processes the labwork summary data received from the backend, adding step grouping.
+ * @param fmsSummary FMSLabworkSummary
+ * @returns LabworkSummary
+ */
+export function processFMSLabworkSummary(fmsSummary: FMSLabworkSummary): LabworkSummary {
+	
+	const result: LabworkSummary = {
+		protocols: []
+	}
+
+	for (const protocolID in fmsSummary.protocols) {
+		const fmsProtocol = fmsSummary.protocols[protocolID]
+		
+		const protocol : LabworkSummaryProtocol = {
+			id: Number.parseInt(protocolID),
+			name: fmsProtocol.name,
+			count: fmsProtocol.count,
+			groups: []
+		}
+
+		const steps = fmsProtocol.steps.map(fmsStep => {
+			const step: LabworkSummaryStep = {
+				id: fmsStep.id,
+				name: fmsStep.name,
+				count: fmsStep.count,
+				specifications: [...fmsStep.step_specifications]
+			}
+			return step
+		})
+
+		// Hard-coded grouping
+		if (protocol.name === 'Library Preparation') {
+
+			// Create a group for each platform
+			const groups = new Map<string, LabworkStepGroup>()
+			groups.set('UNKNOWN', {
+				defaultGroup: false,
+				name: 'UNKNOWN',
+				steps: []
+			})
+
+			// Create a group for each unique platform found in step specs
+			for(const step of steps) {
+				let group
+				const platform = getSpecifiedValue(step, 'Library Platform')
+				if (platform) {
+					if (!groups.has(platform)) {
+						groups.set(platform, {
+							defaultGroup: false,
+							name: platform,
+							steps: []
+						})
+					}
+					group = groups.get(platform)
+				} else {
+					group = groups.get('UNKNOWN')
+				}
+				if (group) {
+					group.steps.push(step)
+				}
+			}
+
+			// Sort the groups by group name (since the order of items in the map is random)
+			const sortedGroups =  [...groups.values()]
+				.sort((groupA, groupB) => {
+					if (!groupA.name || !groupB.name) {
+						return 0
+					}
+					return groupA.name.localeCompare(groupB.name)
+				})
+
+			for (const group of sortedGroups) {
+				addGroupIfNotEmpty(protocol, group)
+			}
+		} else {
+			const defaultGroup : LabworkStepGroup = {
+				defaultGroup: true,
+				steps
+			}
+			addGroupIfNotEmpty(protocol, defaultGroup)
+		}		
+
+		result.protocols.push(protocol)
+
+		// TODO : sort the protocols? By which criteria? Ideally it would match the basic
+		// order of steps in the workflows somehow.
+	}
+
+	return result
+}
+
+function getSpecifiedValue(step: LabworkSummaryStep, displayName: string) {
+	const spec = step.specifications.find(spec => spec.display_name === displayName)
```

Comment:
The idea behind the display name of the specifications was to add columns to the table on the fly using this as column name. I saw you decided to predefine columns in the frontend by protocol. Are these specification information really useful in that context ?

Reply:
I would rather have some kind of identifier key in the step specification (eg. "LIBRARY_PLATFORM") than having to use the display_name field ("Library Platform") to determine the platform for a step, so that we can change the display names in the future without breaking any code (especially if we need to translate the app to another language).

Reply:
GeleHomme would indeed be nice ...